### ISSUE TITLE
Varnish Turpentine (Bug or incorrect config) If-Modified-Since always blank

### ISSUE THREAD

**Issue by itssudeshsahu** [nexcess/magento-turpentine#1394]
Hello,

Can you please help why If-Modified-Since HTTP header is not decting.

Other possible relevant information: CentOS release 6.4, Magento ver. 1.7.0.2, Nginx-1.10.3-1. Varnish-3.0.3-1, Varnish Turpentine 0.5.4. We are using default.vcl which is genearted by varnish turpentine module.

Below is the varnish log

[root@somedomain.com var]# varnishlog -b
   15 BackendOpen  b default 127.0.0.1 58116 127.0.0.1 8080
   15 TxRequest    b GET
   15 TxURL        b /turpentine/esi/getBlock/method/ajax/access/private/ttl/0/hmac/2ffd905a8878cfc0d5224b1b7fbff8038f3ef66e424f055d13a64add9a427b51/data/vfkAkgtv-6qgh7-CGtmVoaj9XoLwKN1zFPOjcC96vNlK6l895isT5keyUx1x3fu-fSjhrFaoNoYyEMygiBoYkY.VPdE.yEnzeHTYSM63lvsLQkpsLkK2374ndq
   15 TxProtocol   b HTTP/1.1
   15 TxHeader     b Host: www.somedomain.com
   15 TxHeader     b User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:52.0) Gecko/20100101 Firefox/52.0
   15 TxHeader     b Accept: text/javascript, text/html, application/xml, text/xml, */*
   15 TxHeader     b Accept-Language: en-GB,en;q=0.5
   15 TxHeader     b Referer: http://www.somedomain.com/services/training.html
   15 TxHeader     b X-NewRelic-ID: VQUEWFBXDRACUVFWAwEDXg==
   15 TxHeader     b X-Requested-With: XMLHttpRequest
   15 TxHeader     b X-Prototype-Version: 1.7
   15 TxHeader     b Cookie: frontend=3a7f41147faa4faa89462b1859e3d1c1; _ga=GA1.2.1187386220.1491721772; wcsid=KCXtgc5TlaTTG3vy845fQ0J0jAqK60Ko; hblid=y76DlX0vzL4Greqr845fQ0J0Ao1A0qKK; _oklv=1491722679329%2CKCXtgc5TlaTTG3vy845fQ0J0jAqK60Ko; _hjIncludedInSample=1; __utma=27257
   15 TxHeader     b X-Forwarded-For: 124.123.79.91
   15 TxHeader     b X-Normalized-User-Agent: firefox
   15 TxHeader     b X-Turpentine-Secret-Handshake: 1
   15 TxHeader     b X-Varnish-Esi-Method: ajax
   15 TxHeader     b X-Varnish-Esi-Access: private
   15 TxHeader     b X-Varnish: 1740877448
   15 TxHeader     b Accept-Encoding: gzip
   15 RxProtocol   b HTTP/1.1
   15 RxStatus     b 200
   15 RxResponse   b OK
   15 RxHeader     b Server: nginx
   15 RxHeader     b Date: Sun, 09 Apr 2017 07:25:14 GMT
   15 RxHeader     b Content-Type: text/html; charset=UTF-8
   15 RxHeader     b Transfer-Encoding: chunked
   15 RxHeader     b Connection: keep-alive
   15 RxHeader     b Expires: Thu, 19 Nov 1981 08:52:00 GMT
   15 RxHeader     b Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
   15 RxHeader     b Pragma: no-cache
   15 RxHeader     b Set-Cookie: frontend=3a7f41147faa4faa89462b1859e3d1c1; expires=Sun, 09-Apr-2017 08:25:14 GMT; Max-Age=3600; path=/; domain=www.somedomain.com
   15 RxHeader     b Access-Control-Allow-Origin: http://www.somedomain.com
   15 RxHeader     b X-Turpentine-Flush-Events: customer_login,customer_logout,sales_quote_save_after,checkout_cart_product_update_after,checkout_cart_save_after,checkout_cart_product_add_after
   15 RxHeader     b X-Turpentine-Cache: 0
   15 RxHeader     b X-Turpentine-Esi: 0
   15 RxHeader     b X-NewRelic-App-Data: PxQGUlVaDgcFR1NXBwUDVlAJFB9AMQYAZBBZDEtZV0ZaClc9HiJGERBZWj1JF01BQAEPFl5fUjpLVEVaPkkEB0QgVA1bUxYeA0sJUQFRA05UGAZSVVAOHx1VTUAPBVVWVgMPAFoCVVsJAwwHGhRSU18WXDw=
   15 RxHeader     b Content-Encoding: gzip
   15 RxHeader     b Vary: Accept-Encoding
   15 RxHeader     b Cache-Control: private
   15 Fetch_Body   b 3(chunked) cls 0 mklen 1
   15 Length       b 476
   15 BackendReuse b default

Regards,
Sudesh

**Comment by itssudeshsahu** [comment]
Sorry varnish turpentine version is 0.6.1.

Regards,
Sudesh

**Comment by miguelbalparda** [comment]
I don't think this is related to Turpentine since as far as I can tell it is not being unset in the default VCL. Is there anything else between Varnish and the browser? [This](http://stackoverflow.com/questions/21094228/varnish-bug-or-incorrect-config-if-none-match-and-if-modified-since-always-bla) might be helpful. It seems you are using nginx.
